### ISSUE TITLE
Clean up Dockerfile / dependencies / environments

### ISSUE THREAD

**Issue by dconathan** [nextml/NEXT#191]
It always kind of bothered me that the backend, workers and mongodb all used the same docker image - seems very messy and defeats the point of docker (for example, workers don't *need* mongo installed, just pymongo, and likewise mongodb doesn't need python).  I believe the only reason for this was as a shortcut/so the mongo_backup could run the backup script, but we're removing that as per #168?

I also ran into some issues here at amfam with the OLD version of pip that is in the ubuntu:14.04 repos (updating it wouldn't even work - something to do with the firewall)... so I decided to take this on.

I changed it so that the mongodb container just uses the `mongo` dockerhub image, and my backend/worker Dockerfile looks like:

```
FROM python:2
MAINTAINER Devin Conathan, ***@***.com

# Install python dependencies for next_backend
ADD requirements.txt /requirements.txt
RUN pip install -r requirements.txt
```

Super simple and easier to maintain.  We could even freeze things at more specific versions (which we probably should be doing anyway).  All tests pass with my setup though cadvisor isn't working properly but seems like a simple fix once I diagnose it.

Thoughts?

**Comment by liamim** [comment]
This is definitely the Right Way™ to do this. I really hope we can get this in master.
It would also be cool if we could use `-alpine` containers since they're far smaller/lighter than the Ubuntu/Debian-based ones.

**Comment by dconathan** [comment]
FYI the cadvisor box is related to https://github.com/google/cadvisor/issues/1556 - I think it's because the mongo dockerhub image is built on busybox.  We should either use a different mongo docker image or wait for this bug to get fixed.

@liamim re: alpine containers from what I've heard it's not worth the extra maintenance once you are doing stuff with e.g. numpy - once you add all the tools/libs necessary to compile, the alpine image is about the same size as a debian image with numpy.

**Comment by lalitkumarj** [comment]
I approve.

**Comment by liamim** [comment]
@dconathan that's a fair point! in any case, having up-to-date libraries in the apt repositories would be nice.

**Comment by liamim** [comment]
This was fixed by #195; closing.
